lib/p2p/packets/types/DisconnectingPacket.ts

```diff
@@ -0,0 +1,31 @@
+import Packet, { PacketDirection } from '../Packet';
+import PacketType from '../PacketType';
+
+export type DisconnectingPacketBody = {
+  reason: DisconnectionReason;
+  payload?: string;
+};
+
+export enum DisconnectionReason {
+  ResponseStalling,
+  IncompatibleProtocolVersion,
+  UnexpectedIdentity,
+  ForbiddenIdentityUpdate,
+  ConnectedToSelf,
+  NotReadyForConnections,
```

Comment:
I think this should be something like "NotAcceptingConnections" - because "not ready" implies that we will be ready in the future, which is not always going to be the case. For now the only time we're using this is when we're in the process of shutting down the p2p pool.